src/hotspot/share/opto/loopnode.cpp

```diff
@@ -4745,78 +4795,77 @@ void PhaseIdealLoop::verify_compare( Node *n, const PhaseIdealLoop *loop_verify,
     loop_verify->idom_no_update(n)->dump();
     tty->cr();
   }
-
+  */
 }
 
 //------------------------------verify_tree------------------------------------
-// Verify that tree structures match.  Because the CFG can change, siblings
-// within the loop tree can be reordered.  We attempt to deal with that by
+// Verify that tree structures match. Because the CFG can change, siblings
+// within the loop tree can be reordered. We attempt to deal with that by
 // reordering the verify's loop tree if possible.
-void IdealLoopTree::verify_tree(IdealLoopTree *loop, const IdealLoopTree *parent) const {
-  assert( _parent == parent, "Badly formed loop tree" );
+void IdealLoopTree::verify_tree(IdealLoopTree* loop, const IdealLoopTree* parent) const {
+  assert(_parent == parent, "Badly formed loop tree");
 
-  // Siblings not in same order?  Attempt to re-order.
-  if( _head != loop->_head ) {
-    // Find _next pointer to update
+  // If "this" and "loop" are not the same sibling of "parent", reorder the siblings.
+  if (_head != loop->_head) {
+    tty->print_cr("reorder loop tree");
+    // Find _next pointer to update (where "loop" is attached)
     IdealLoopTree **pp = &loop->_parent->_child;
-    while( *pp != loop )
+    while (*pp != loop) {
       pp = &((*pp)->_next);
```

Comment:
May be instead of this convoluted search, swap and verification code we can simply reorder whole siblings list first before verification? You don't need to swap - just create local ordered list.

Reply:
Ok, I will refactor this code. I will have two local lists, sort them by `_head->_idx`. Then I can also verify that the lists are the same.